src/core/server/saved_objects/migrationsv2/next.ts

```diff
@@ -83,41 +91,48 @@ export const nextActionMap = (client: ElasticsearchClient, transformRawDocs: Tra
          * Since we don't run a search against the target index, we disable "refresh" to speed up
          * the migration process.
          * Although any further step must run "refresh" for the target index
-         * before we reach out to the OUTDATED_DOCUMENTS_SEARCH step.
-         * Right now, we rely on UPDATE_TARGET_MAPPINGS + UPDATE_TARGET_MAPPINGS_WAIT_FOR_TASK
-         * to perform refresh.
+         * before we reach out to the OUTDATED_DOCUMENTS_SEARCH_OPEN_PIT step.
+         * Right now, it's performed during REFRESH_TARGET step.
          */
         false
       ),
     SET_TEMP_WRITE_BLOCK: (state: SetTempWriteBlock) =>
       Actions.setWriteBlock(client, state.tempIndex),
     CLONE_TEMP_TO_TARGET: (state: CloneTempToSource) =>
       Actions.cloneIndex(client, state.tempIndex, state.targetIndex),
+    REFRESH_TARGET: (state: RefreshTarget) => Actions.refreshIndex(client, state.targetIndex),
     UPDATE_TARGET_MAPPINGS: (state: UpdateTargetMappingsState) =>
       Actions.updateAndPickupMappings(client, state.targetIndex, state.targetIndexMappings),
     UPDATE_TARGET_MAPPINGS_WAIT_FOR_TASK: (state: UpdateTargetMappingsWaitForTaskState) =>
       Actions.waitForPickupUpdatedMappingsTask(client, state.updateTargetMappingsTaskId, '60s'),
-    OUTDATED_DOCUMENTS_SEARCH: (state: OutdatedDocumentsSearch) =>
-      Actions.searchForOutdatedDocuments(client, {
-        batchSize: state.batchSize,
-        targetIndex: state.targetIndex,
-        outdatedDocumentsQuery: state.outdatedDocumentsQuery,
-      }),
+    OUTDATED_DOCUMENTS_SEARCH_OPEN_PIT: (state: OutdatedDocumentsSearchOpenPit) =>
+      Actions.openPit(client, state.targetIndex),
+    OUTDATED_DOCUMENTS_SEARCH_READ: (state: OutdatedDocumentsSearchRead) =>
+      Actions.readWithPit(
+        client,
+        state.pitId,
+        state.outdatedDocumentsQuery,
```

Comment:
it doesn't use `unusedTypesQuery` because `target index` already contains filtered SO types